odo/backends/tests/test_sql.py

```diff
@@ -86,10 +88,10 @@ def test_discover():
                  sa.Column('name', sa.String),
                  sa.Column('amount', sa.Integer),
                  sa.Column('timestamp', sa.DateTime, primary_key=True))
-
-    assert discover(s) == \
-        dshape('var * {name: ?string, amount: ?int32, timestamp: datetime}')
-
+    ds = dshape('var * {name: ?string, amount: ?int32, timestamp: datetime}')
+    assert discover(s) == ds
```

Comment:
since this test was added we have added `datashape.util.assert_dshape_equal`. This function provides much more helpful error messages in the failure case, for example, imagine the following two moderatly complex dshapes:

```python
a = dshape('var * {a: {b: ?int32, c: ?float64}, d: string[10]}')
b = dshape('var * {a: {b: ?int32, c: ?float32}, d: string[10]}')
```

These are slightly different, with assert a == b we get:

```
    def test_ex():
        a = dshape('var * {a: {b: ?int32, c: ?float64}, d: string[10]}')
        b = dshape('var * {a: {b: ?int32, c: ?float32}, d: string[10]}')
>       assert a == b
E       assert dshape("var *... string[10]}") == dshape("var * ... string[10]}")
E         Use -v to get the full diff

```

with `assert_dshape_equal` we get the path right to the error:
```
E           AssertionError: ctype names do not match: 'float64' != 'float32'
E           path: _.measure['a']['c'].ty.name
```

We can also pass a bunch of special arguments to change certain comparisons like ignoring the order of records or ignoring fixed lengths for strings.